runtime/astar/src/xcm_config.rs

```diff
@@ -61,21 +60,40 @@ pub type CurrencyTransactor = CurrencyAdapter<
     (),
 >;
 
-pub struct AsAssetWithRelay<AssetId>(PhantomData<AssetId>);
-impl<AssetId> xcm_executor::traits::Convert<MultiLocation, AssetId> for AsAssetWithRelay<AssetId>
-where
-    AssetId: Clone + Eq + Bounded,
-{
+pub struct AsAssetWithRelay();
+impl xcm_executor::traits::Convert<MultiLocation, AssetId> for AsAssetWithRelay {
     fn convert_ref(id: impl Borrow<MultiLocation>) -> Result<AssetId, ()> {
-        if id.borrow().eq(&MultiLocation::parent()) {
-            Ok(AssetId::max_value())
-        } else {
-            Err(())
+        match id.borrow() {
+            // Native relaychain asset
+            MultiLocation {
+                parents: 0,
+                interior: Junctions::Here,
+            } => Ok(AssetIdHelper::RELAY_CHAIN_ASSET_ID),
+            // Statemint `pallet_assets` assets
+            MultiLocation {
+                parents: 1,
+                interior: X3(Parachain(1000), PalletInstance(50), GeneralIndex(index)),
```

Comment:
So, may be we could have constant or helper function (`statemine_asset_location(index)`) for it, magic numbers anti-pattern.

Reply:
Oh sure, I added it for unit tests actually (didn't see your comment) but I will make it non-UT specific.

Reply:
@akru regarding const/function approach, I cannot use pattern matching then.

How about I just make `1000` and `50` constants?

Reply:
It works up to me